### PR TITLE
Added ldap helper

### PR DESCRIPTION
**Pull request by SimRoberts9**

### CODE REVIEW

File: tests/Unit/Helpers/LdapHelper/SearchByNameTest.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+use NetworkRailBusinessSystems\UserLogin\Helpers\LdapHelper;
+use NetworkRailBusinessSystems\UserLogin\Tests\TestCase;
+
+class SearchByNameTest extends TestCase
+{
+    public function testFindsByName(): void
+    {
+        $this->mock('alias:LdapRecord\Models\ActiveDirectory\User', function ($mock) {
+            $mock->shouldReceive('query->where->orWhere->andFilter->select->limit->get')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/SearchByNameTest.php:13]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/SearchByEmailTest.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+use NetworkRailBusinessSystems\UserLogin\Helpers\LdapHelper;
+use NetworkRailBusinessSystems\UserLogin\Tests\TestCase;
+
+class SearchByEmailTest extends TestCase
+{
+    public function testFindsEmails(): void
+    {
+        $this->mock('alias:LdapRecord\Models\ActiveDirectory\User', function ($mock) {
+            $mock->shouldReceive('query->where->andFilter->select->limit->get')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/SearchByEmailTest.php:13]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+use NetworkRailBusinessSystems\UserLogin\Helpers\LdapHelper;
+use NetworkRailBusinessSystems\UserLogin\Tests\Models\User;
+use NetworkRailBusinessSystems\UserLogin\Tests\TestCase;
+
+class ImportTest extends TestCase
+{
+    public function testThrowsExceptionWhenNotInDirectory(): void
+    {
+        $this->mock('alias:LdapRecord\Models\ActiveDirectory\User', function ($mock) {
+            $mock->shouldReceive('query->where->andFilter->select->limit->get')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                ->andReturn([]);
+        });
+
+        $this->mock('overload:Illuminate\Support\Facades\Artisan', function ($mock) {
+            $mock->shouldReceive('call')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        $this->assertEquals(
+            'Import cancelled; no User was found with the e-mail "ringwraith@example.com" in Active Directory',
+            LdapHelper::import('ringwraith@example.com'),
+        );
+    }
+
+    public function testImportsUser(): void
+    {
+        $this->mock('alias:LdapRecord\Models\ActiveDirectory\User', function ($mock) {
+            $mock->shouldReceive('query->where->andFilter->select->limit->get')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                ->andReturn([Model::class]);
+        });
+
+        $this->mock('overload:Illuminate\Support\Facades\Artisan', function ($mock) {
+            $mock->shouldReceive('call')
+                ->andReturnUsing(function () {
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        $this->assertEquals(
+            'peregrin.took@example.com',
+            LdapHelper::import('peregrin.took@example.com')->email,
+        );
+    }
+
+    public function testThrowsExceptionWhenUserMissing(): void
+    {
+        $this->mock('alias:LdapRecord\Models\ActiveDirectory\User', function ($mock) {
+            $mock->shouldReceive('query->where->andFilter->select->limit->get')
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                ->andReturn([Model::class]);
+        });
+
+        $this->mock('overload:Illuminate\Support\Facades\Artisan', function ($mock) {
+            $mock->shouldReceive('call')
+                ->andReturnUsing(function () {
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Add `once` to the chain to formalise the assertion.

File: tests/Unit/Helpers/LdapHelper/ImportTest.php
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php:1]: DRY. Add helper function to define mocks instead of repeating yourself.

**Review by AnthonyEdmonds — CHANGES_REQUESTED** [review]

File: tests/Unit/Helpers/LdapHelper/ImportTest.php [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+use NetworkRailBusinessSystems\UserLogin\Tests\Models\User;
+use NetworkRailBusinessSystems\UserLogin\Tests\TestCase;
+
+class ImportTest extends TestCase
+{
+    public function testThrowsExceptionWhenNotInDirectory(): void
+    {
+        $this->runMock(true);
+
+        $this->expectException(\ErrorException::class);
```
Comment (AnthonyEdmonds) [inline tests/Unit/Helpers/LdapHelper/ImportTest.php]: Import class

**Review by AnthonyEdmonds — CHANGES_REQUESTED** [review]

File: tests/Unit/Helpers/LdapHelper/ImportTest.php (hunk 9 above)
Comment (SimRoberts9) [reply tests/Unit/Helpers/LdapHelper/ImportTest.php]: Done, thank you!

**Review by AnthonyEdmonds — APPROVED** [review]

**Pull request merged**